File: InvokingParameterizedPythonMethodFromJava/pythonMethodFromJava.java
package InvokingParameterizedPythonMethodFromJava;
// package py4j.examples;
import java.util.ArrayList;
import py4j.GatewayServer;

public class pythonMethodFromJava {
    public void ListOperation(pythonMethodFromJava op) {
        ArrayList <Integer> numbers = new ArrayList<Integer>();
        numbers.add(1);
        numbers.add(2);
        numbers.add(3);
        ArrayList<Integer> result = op.Update(numbers);
        System.out.println("Updated list :");
        for (int i =0; i < result.size(); i++ ) {
            System.out.print(result.get(i));
            System.out.print(" ");
        }
    }
    public static void main(String[] args) {
        GatewayServer gatewayServer = new GatewayServer(new pythonMethodFromJava());
        gatewayServer.start();
        System.out.println("gateway server started");
    }
}

File: InvokingParameterizedPythonMethodFromJava/pythonMethodFromJava.py
from py4j.java_gateway import JavaGateway, GatewayParameters

class ListManipulation(object):
    def Update(self, lst):
        for i in range(0, len(lst)):
            lst[i] = lst[i] * 2
        print(lst)
        return lst
    
    class java:
        implements = ['py4j.InvokingParameterizedPythonMethodFromJava.pythonMethodFromJava']

if __name__=='__main__':
    gateway = JavaGateway(start_callback_server=True, gateway_parameters=GatewayParameters(auto_convert=True))
    lst = ListManipulation()
    javaClasslst = gateway.jvm.py4j.examples.pythonMethodFromJava()
    # pass python object to the java program
    javaClasslst.ListOperation(lst)